azure-kusto-data/azure/kusto/data/request.py

```diff
@@ -356,23 +345,22 @@ def execute_streaming_ingest(
             request_params["mappingName"] = mapping_name
 
         request_headers = {
-            "Accept-Encoding": accept_encoding,
-            "Connection": connection,
+            "Accept": "application/json",
+            "Accept-Encoding": "gzip,deflate",
             "x-ms-client-version": "Kusto.Python.StreamingClient:" + VERSION,
             "x-ms-client-request-id": "KPSC.execute;" + str(uuid.uuid4()),
-            "Host": self._streaming_ingest_endpoint.split("/")[2],
         }
 
-        if accept is not None:
-            request_headers["Accept"] = accept
-        if content_encoding is not None:
-            request_headers["Content-Encoding"] = content_encoding
-        if expect is not None:
-            request_headers["Expect"] = expect
-        if content_length is not None:
-            request_headers["Content-Length"] = str(content_length)
         if self._auth_provider:
             request_headers["Authorization"] = self._auth_provider.acquire_authorization_header()
+        if client_request_properties.has_option("Connection"):

```

Comment:
All of those should come from the client request properties.
This is why the to_json method is exposed there.